pydicom/encaps.py

```diff
@@ -370,3 +379,207 @@ def read_item(fp):
+
+    Parameters
+    ----------
+    frame : bytes
+        The data to fragment.
+    no_fragments : int, optional
+        The number of fragments (default 1).
+
+    Yields
+    ------
+    bytes
+        The fragmented data, with all fragments as an even number of bytes
+        greater than or equal to two.
+
+    Notes
+    -----
+
+    * All items containing an encoded fragment shall be made of an even number
+      of bytes greater than or equal to two.
+    * The last fragment of a frame may be padded, if necessary to meet the
+      sequence item format requirements of the DICOM Standard.
+    * Any necessary padding may be appended after the end of image marker.
+    * Encapsulated Pixel Data has the Value Representation OB.
+    * Values with a VR of OB shall be padded with a single trailing NULL byte
+      value (0x00) to achieve even length.
+
+    References
+    ----------
+    DICOM Standard, Part 5, Section 6.2 and Annex A.4
+    """
+    frame_length = len(frame)
+    if no_fragments > frame_length:
+        raise ValueError('The number of fragments is larger than the '
+                         'number of bytes in the frame.')
+
+    # We use floor to ensure we don't end up with too many fragments
+    length = int(floor(frame_length / no_fragments))
+
+    # Each item shall be an even number of bytes
+    if length % 2:
+        length += 1
+
+    offset = 0
+    for ii in range(no_fragments):
+        if ii < (no_fragments - 1):
+            # 1st to (N-1)th fragment
+            fragment = frame[offset:offset + length]
+        else:
+            # Nth fragment
+            fragment = bytearray(frame[offset:])
+
+            # Pad last fragment if needed to make it even
+            if (frame_length - offset) % 2:
+                fragment.extend(b'\x00')
+
+        offset += length
+        yield bytes(fragment)
+
+
+def itemise_fragment(fragment):
+    """Return an itemised `fragment`.
+
+    Parameters
+    ----------
+    fragment : bytes
+        The fragment to itemise.
+
+    Returns
+    -------
+    bytes
+        The itemised fragment.
+
+    Notes
+    -----
+
+    * The encoding of the shall be in Little Endian.
+    * Each fragment is encapsulated as a DICOM Item with tag (FFFE,E000), then
+      a 4 byte length.
+    """
+    item = bytearray()
+    # item tag (fffe,e000)
+    item.extend(b'\xFE\xFF\x00\xE0')
+    # fragment length '<I' little endian, 4 byte unsigned int
+    item.extend(pack('<I', len(fragment)))
+    # fragment data
+    item.extend(fragment)
+
+    return bytes(item)
+
+
+itemize_fragment = itemise_fragment
+
+
+def itemise_frame(frame, no_fragments=1):
+    """Yield items generated from `frame`.
+
+    Parameters
+    ----------
+    frame : bytes
+        The data to fragment and itemise.
+    no_fragments : int, optional
+        The number of fragments/items (default 1).
+
+    Yields
+    ------
+    bytes
+        An itemised fragment, encoded as little endian.
```

Comment:
fragment ->frame

Reply:
No, this is correct. The frame is fragmented then itemised and the itemised fragment yielded. I'll reword it a little bit to make it clearer.